### DISCUSSION TITLE
The item database is incompatible, re-creating database table structures

### DISCUSSION THREAD

**Discussion by sitadrost** [abraunegg/onedrive#2870] [Q&A] [answered]
Recently, each time I run onedrive to back up my local files, I'm getting this warning:

`The item database is incompatible, re-creating database table structures`

And then, for every single local file and folder:

`The file we are attempting to upload as a new file already exists on Microsoft OneDrive:`

Apart from this, everything seems to be updated alright, it just takes a lot longer than it used to.

I'm not sure what causes this, so any help would be appreciated!

Details that might be relevant:
Onedrive version v2.5.2
Ubuntu 24.04
Command used:
`onedrive --sync --verbose --upload-only`

**Comment by abraunegg** [comment, marked as answer]
> `The item database is incompatible, re-creating database table structures`

v2.5.x introduces a DB change. This is normal.

If your are using --dry-run at all without updating your DB .. you will *always* get this message.

> Apart from this, everything seems to be updated alright, it just takes a lot longer than it used to.

As per the release notes for v2.5.0 a first time sync with v2.5.x will take longer - due to many factors.

Apart from that - everything is normal. No issue for you except your use of Ubuntu ... thats your choice.

**Reply by sitadrost** [reply]
Thanks, that's good to know. 
I've been using v2.5.x for a while now though (I always update before running onedrive), is it really going to re-create those database table structures EACH TIME I run onedrive from now on?

**Reply by abraunegg** [reply]
> I've been using v2.5.x for a while now though (I always update before running onedrive), is it really going to re-create those database table structures EACH TIME I run onedrive from now on?

No - it should not.

There was a schema change from v2.4.25 --> v2.5.0 and another from v2.5.0 --> v2.5.2 due to the timestamp issue caused by v2.5.1.

If the client is - even now - re-creating the database each time you run it when running v2.5.2 ... then there is something odd going on with your platform and/or where the system is storing the database.

**Reply by sitadrost** [reply]
Ah, right, thanks. I'll see what happens next time I run v2.5.2; hopefully it will run normally then.
